### PR TITLE
Add entry point flag #36

### PR DESCRIPTION
**Pull request by AaronStGeorge**
Example use of new flag:
```
aaron@ralph4:~/Dev/c++/spf-ie$ ./build/bin/spf-ie test/simple_two_funcs.c --print-info
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "test/simple_two_funcs.c"
No compilation database found in /home/aaron/Dev/c++/spf-ie/test or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.

Processing: /home/aaron/Dev/c++/spf-ie/test/simple_two_funcs.c
=================================================

FUNCTION: func_1
---------------

Statements:
S0: *$x$ = $a$ + $b$;

Iteration spaces:
S0: { [0] }

Execution schedules:
S0: { [0] -> [0] }

Data spaces: {$a$ int, $b$ int, $x$ int *}

Array reads:
S0:{
    $a$: { [0] -> [0] }
    $b$: { [0] -> [0] }
}

Array writes:
S0: none

FUNCTION: func_2
---------------

Statements:
S0: *$x$ = $a$ + $b$;

Iteration spaces:
S0: { [0] }

Execution schedules:
S0: { [0] -> [0] }

Data spaces: {$a$ int, $b$ int, $x$ int *}

Array reads:
S0:{
    $a$: { [0] -> [0] }
    $b$: { [0] -> [0] }
}

Array writes:
S0: none

aaron@ralph4:~/Dev/c++/spf-ie$ ./build/bin/spf-ie test/simple_two_funcs.c --print-info --entry-point func_2
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "test/simple_two_funcs.c"
No compilation database found in /home/aaron/Dev/c++/spf-ie/test or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.

Processing: /home/aaron/Dev/c++/spf-ie/test/simple_two_funcs.c
=================================================

FUNCTION: func_2
---------------

Statements:
S0: *$x$ = $a$ + $b$;

Iteration spaces:
S0: { [0] }

Execution schedules:
S0: { [0] -> [0] }

Data spaces: {$a$ int, $b$ int, $x$ int *}

Array reads:
S0:{
    $a$: { [0] -> [0] }
    $b$: { [0] -> [0] }
}

Array writes:
S0: none

aaron@ralph4:~/Dev/c++/spf-ie$
```

I did notice what was maybe an existing bug when running with `test/nesting_test.c` example when running on a build from the `main` branch:

```
aaron@ralph4:~/Dev/c++/spf-ie$ ./build/bin/spf-ie test/nesting_test.c --print-info
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "test/nesting_test.c"
No compilation database found in /home/aaron/Dev/c++/spf-ie/test or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.

Processing: /home/aaron/Dev/c++/spf-ie/test/nesting_test.c
=================================================

FUNCTION: asdf
---------------

Statements:
S0: int i;
S1: int x;
S2: $x__w__0$ = i;
S3: $x$ = $x__w__0$ + 5;

Iteration spaces:
S0: { [0] }
S1: { [0] }
S2: { [i] : i >= 0 && -i + 2 >= 0 }
S3: { [i] : i >= 0 && -i + 2 >= 0 && i - 2 >= 0 }

Execution schedules:
S0: { [0] -> [0] }
S1: { [0] -> [1] }
S2: { [i] -> [2, i, 0] : i - i = 0 }
S3: { [i] -> [2, i, 1] : i - i = 0 }

Data spaces: {$a$ int, $x$ int, $x__w__0$ int}

Array reads:
S0: none
S1: none
S2: none
S3:{
    $x__w__0$: { [i] -> [0] }
}

Array writes:
S0: none
S1: none
S2:{
    $x__w__0$: { [i] -> [0] }
}
S3:{
    $x$: { [i] -> [0] }
}

FUNCTION: main
---------------

ERROR: Found a statement following a return statement. Returns are only allowed at the end of functions.
At /home/aaron/Dev/c++/spf-ie/test/nesting_test.c:14:5:
int i;

aaron@ralph4:~/Dev/c++/spf-ie$ 

```

### CODE REVIEW

**Comment by riftEmber** [comment]
Oh yep, that looks like a bug. I was kind of imagining this flag to be required so we always just process a single function (which would make that issue irrelevant), but that's also not really a requirement -- what do you think about it @AaronStGeorge ?

**Comment by AaronStGeorge** [comment]
@riftEmber I don't have an opinion on making the flag required. You would be the expert :). I'll push a change to do that soon.

I'm not sure what's up with CI at the moment, it builds and tests pass locally. Noticing this in the logs:
```
21:12:18  Error: 1 error occurred:
21:12:18  	* unable to find 'registry.hub.docker.com/riftember/spf-ie:jenkins-spf-ie-entry-point-2' in local storage: no such image
```

**Comment by riftEmber** [comment]
I guess I would be in favor of making it required then, just because as we go towards spf-ie taking code through the whole pipeline it doesn't make sense to do that separately for several functions.

As for the CI, don't worry about that since it's just my experiment and breaks frequently. I see that the pr-head build (which builds your actual PR code) succeeded, whereas the pr-merge build (which tests out building your code if it were merged to main) failed, but the error it gave seems to be with my system setup, not the code.

**Comment by riftEmber** [comment]
In this PR could you also add an argument to the `build computation from code` helper method (in the test file) that specifies name of function to process? And adjust existing calls of it to use that

File: src/Driver.cpp
Comment (riftEmber) [inline src/Driver.cpp:49]: what systems will this work properly on? (I assume that's a color specification)

Comment (AaronStGeorge) [reply src/Driver.cpp:49]: It is.

From [wikipedia](https://en.wikipedia.org/wiki/ANSI_escape_code#Platform_support):

> [Unix-like systems] Terminal emulators for communicating with local programs as well as remote machines and the text system console almost always support ANSI escape codes. This includes terminal emulators such as xterm, rxvt, GNOME Terminal, and Konsole on systems with X11-based or Wayland-based window systems, and Terminal.app and third-party terminal emulators such as iTerm2 on macOS.

> [Windows] In 2016, Microsoft released the Windows 10 version 1511 update which unexpectedly implemented support for ANSI escape sequences, over two decades after the debut of Windows NT.[13] This was done alongside Windows Subsystem for Linux, allowing Unix-like terminal-based software to use the sequences in Windows Console. Unfortunately this defaults to off, but Windows PowerShell 5.1 enabled it. PowerShell 6 made it possible to embed the necessary ESC character into a string with `e.[14] Windows Terminal, introduced in 2019, supports the sequences by default, and Microsoft intends to replace the Windows Console with Windows Terminal.[15]


Judging by that I would say essentially everywhere :shrug: .

Comment (riftEmber) [reply src/Driver.cpp:49]: Ok cool, I just remembered from a few years back when I tried to do something like this it was very system specific (maybe I used a different method). But this seems universal enough, and it works on my zsh and presumably your bash so :+1:

**Review by riftEmber — APPROVED** [review]

**Comment by riftEmber** [comment]
So for merging, I typically squash only if there are commits that don't make sense in the full context of the project. In your case, you have the "code reviews" commit, so you might either want to squash that into the previous commit which references the issue, or reword it to what functionality it changes if you feel like it makes sense as it's own commit.

**Comment by AaronStGeorge** [comment]
Ok, I just squashed into the `Add entry point flag` commit.

**Comment by riftEmber** [comment]
:+1:

**Comment by AaronStGeorge** [comment]
@riftEmber anything else you wanted to see before merging this?

**Comment by riftEmber** [comment]
nope, looks good to me!

**Comment by AaronStGeorge** [comment]
Ok, I'm not authorized to merge this pull request, so you'll have to merge or add me to whatever list I need to be added to.
image

**Comment by riftEmber** [comment]
I forgot this project wasn't open for anyone in adapt to write, I added you to write access now

**Comment by AaronStGeorge** [comment]
image
Hmm I still can't merge. I imagine I'm not on [some protected branch list](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#restrict-who-can-push-to-matching-branches) maybe :shrug:?

**Comment by riftEmber** [comment]
Whoops, looks like I left an "approved push users" whitelist enabled and empty. Try now?

**Pull request merged**
